### PR TITLE
Make UpgradeToPro CTA for pitr and custom domains more nuanced based on the project's plan

### PR DESCRIPTION
**Pull request by joshenlim**
If on free plan
image

If already on paid plan
image

### CODE REVIEW

File: studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx [hunk 1]
```diff
@@ -49,7 +52,11 @@ const CustomDomainConfig = () => {
                 icon={<IconAlertCircle size={18} strokeWidth={1.5} />}
                 primaryText="Custom domains are a Pro plan add-on"
                 projectRef={ref as string}
-                secondaryText="To configure a custom domain for your project, please upgrade to a Pro plan"
+                secondaryText={
+                  tier === PRICING_TIER_PRODUCT_IDS.FREE
+                    ? 'To configure a custom domain for your project, please upgrade to the Pro plan and purchase the add-on'
```
Comment (alaister) [inline studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx:57]: "To configure a custom domain for your project, please upgrade to the Pro plan _with the custom domains add-on selected_"

You purchase the add-on in the same step as upgrading to pro, right? Currently, it sounds like two different steps.
Feel free to change my suggestion if you think there's an even clearer way of explaining it :)

File: studio/pages/project/[ref]/database/backups/pitr.tsx [hunk 2]
```diff
@@ -63,8 +65,12 @@ const PITR = () => {
     return isPITRSelfServeEnabled ? (
       <UpgradeToPro
         projectRef={ref}
-        primaryText="Free Plan does not include point in time recovery."
-        secondaryText="Please upgrade to the Pro plan to enable point in time recovery for your project."
+        primaryText="Point in time recovery is a Pro plan add-on."
+        secondaryText={
+          tier === PRICING_TIER_PRODUCT_IDS.FREE
+            ? 'Please upgrade to the Pro plan and purchase the add-on to enable point in time recovery for your project.'
```
Comment (alaister) [inline studio/pages/project/[ref]/database/backups/pitr.tsx:71]: Same as the other instance :)

**Review by alaister — CHANGES_REQUESTED** [review]
Just a couple of wording nits. Feel free to overrule me if you disagree – don't have my heart set haha.

Otherwise, LGTM!

File: studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx:57]: oh yup! that does sound better, thanks man will update

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**